### DISCUSSION TITLE
How to theme nested menus using ItemContainerTheme ?

### DISCUSSION THREAD

**Discussion by flarive** [AvaloniaUI/Avalonia#13425] [General]
Hello,

I'm trying to create a custom theme for a Menu with nested sub menus items.

Here is my menu :
```
<Menu Padding="0" Margin="0" HorizontalAlignment="Center" Theme="{StaticResource MenuCapsuleBarOutset}">
  <MenuItem Height="32" Header="File">
    <MenuItem Header="New">
      <MenuItem Header="Image">
        <MenuItem Header="GIF"/>
        <MenuItem Header="JPG"/>
        <MenuItem Header="PNG"/>
      </MenuItem>
      <MenuItem Header="Video">
        <MenuItem Header="MP4"/>
        <MenuItem Header="MKV"/>
        <MenuItem Header="MOV"/>
      </MenuItem>
      <MenuItem Header="Music">
        <MenuItem Header="MP3"/>
        <MenuItem Header="FLAC"/>
        <MenuItem Header="WAV"/>
      </MenuItem>
    </MenuItem>
    <MenuItem Header="Open"/>
    <MenuItem Header="Save"/>
    <MenuItem Header="Delete"/>
  </MenuItem>
  <MenuItem Height="32" Header="Help">
    <MenuItem Header="About"/>
  </MenuItem>
</Menu>
```

Here is my MenuCapsuleBarOutset theme :
```
  <ControlTheme x:Key="MenuCapsuleBarOutset" TargetType="Menu" BasedOn="{StaticResource NoMaterialMenu}">
    <Setter Property="Background" Value="Transparent"/>
    <Setter Property="Height" Value="48"/>
    <Setter Property="HorizontalAlignment" Value="Left"/>
    <Setter Property="BorderBrush" Value="{DynamicResource MenuFlyoutPresenterBorderBrush}"/>
    <Setter Property="BorderThickness" Value="0"/>
    <Setter Property="ItemContainerTheme" Value="{StaticResource MenuCapsuleBarInsetMenuItem}" />
    <Setter Property="ClipToBounds" Value="False" />

    <Style Selector="^ /template/ Border#PART_BorderLayout">
      <Setter Property="BoxShadow" Value="-5 -5 20 Red, 5 5 20 Blue"/>
      <Setter Property="Height" Value="48"/>
    </Style>
  </ControlTheme>
```

This menu theme is using an ItemContainerTheme named MenuCapsuleBarInsetMenuItem.

Here is my MenuCapsuleBarInsetMenuItem theme :
  ```
<ControlTheme x:Key="MenuCapsuleBarInsetMenuItem" TargetType="MenuItem" BasedOn="{StaticResource NoMaterialTopLevelMenuItem}">
    <Setter Property="assist:MenuAssist.PopupAnchor" Value="TopLeft" />
    <Setter Property="assist:MenuAssist.PopupPlacement" Value="Bottom" />
    <Setter Property="assist:MenuAssist.PopupHorizontalOffset" Value="0" />
    <Setter Property="assist:MenuAssist.PopupVerticalOffset" Value="-10" />
    <Setter Property="assist:MenuAssist.PopupBorderThickness" Value="0" />
    <Setter Property="assist:MenuAssist.PopupBorderBrush" Value="Transparent" />

    <Style Selector="^ /template/ Border#PART_PopupBorder">
      <Setter Property="BoxShadow" Value="-5 -5 20 Red, 5 5 20 Blue"/>
      <Setter Property="Margin" Value="10"/>
    </Style>
  </ControlTheme>
```

How could i have all sub menus themed with a red/blue box shadow and not only the first level sub menu ?
I have tried adding 
`    <Setter Property="ItemContainerTheme" Value="{StaticResource MenuCapsuleBarInsetMenuItem}" />`
in MenuCapsuleBarInsetMenuItem but it doesn't work as expected, i have  strange behavior, the sub menu styles are completelly broken :(

![sample](https://github.com/AvaloniaUI/Avalonia/assets/7724904/844e724f-e583-4a0b-bf86-141896a2bb58)

Here is a ready to use sample repo : https://github.com/flarive/MenuItem_ItemContainerTheme_Sample

Thanks a lot for your help

Eviral

**Comment by flarive** [comment]
Nobody ? @grokys, @kekekeks, @danwalmsley, @maxkatz6 ?
I'm working on Menus Neumorphism theme and this is the last issue i have :)
Thanks for your help
Eviral

**Comment by thevortexcloud** [comment]
Typically you would use a hierarchical template, or in Avalonia's case `TreeDataTemplate`. It works pretty much the same as it does in WPF

https://learn.microsoft.com/en-us/dotnet/api/system.windows.hierarchicaldatatemplate?view=windowsdesktop-7.0

But in the case of themes specifically I can't say. I have never made one.

**Comment by flarive** [comment]
It doesn't help me much, sorry...
TreeDataTemplate doesn't exist for Menu and MenuItem :(

**Comment by flarive** [comment]
I finally found where the problem was !
There is no bug related to ItemContainerTheme in Avalonia 11 :)

I just messed up with templates on my side, i was using root level menu item template for sub levels menu items.
